ext/session/session.c

```diff
@@ -2711,6 +2723,22 @@ static const zend_function_entry session_functions[] = {
 
 /* {{{ SessionHandlerInterface functions[]
 */
+static const zend_function_entry php_session_save_handler_iface_functions[] = {
+	PHP_ABSTRACT_ME(SessionSaveHandlerInterface, open, arginfo_session_class_open)
```

Comment:
This interface seems to be extremely bloated and extremely implementation-detailed (which is why the savehandler interfaces out there are so full of edge cases).

Isn't it simpler to have following?

```php
interface SessionSaveHandlerInterface
{
    /** just save or refresh the damn thing - metadata can contain arbitrary info, such as TTL too */
    public function save(SessionId $id, SessionMetadata $metadata, SessionData $session);
    public function destroy(SessionId $id);
    public function load(SessionId $id);
}
```